lib/atat-iam-stack.ts

```diff
@@ -78,62 +54,38 @@ export class AtatIamStack extends cdk.Stack {
       statements: [
         // During a deployment, outside of CloudFormation (where the role will be used),
         // the CDK needs to read from and write to the S3 buckets for the CDK.
-        new iam.PolicyStatement({
-          sid: "AllowModifyingCdkToolBuckets",
-          effect: iam.Effect.ALLOW,
-          actions: ["s3:*"],
-          resources: [
-            `arn:${cdk.Aws.PARTITION}:s3:::cdk-*-assets-${cdk.Aws.ACCOUNT_ID}-*`,
-            `arn:${cdk.Aws.PARTITION}:s3:::cdktoolkit-stagingbucket-*`,
-          ],
-        }),
-        new iam.PolicyStatement({
-          sid: "AllowAssumingCdkRoles",
-          effect: iam.Effect.ALLOW,
-          actions: ["sts:AssumeRole"],
-          resources: [
-            this.formatArn({
-              service: "iam",
-              region: "",
-              resource: "role",
-              resourceName: `cdk-${DEFAULT_BOOSTRAP_QUALIFIER}-*-role-${cdk.Aws.ACCOUNT_ID}-${cdk.Aws.REGION}`,
-            }),
-          ],
-        }),
-        new iam.PolicyStatement({
-          sid: "AllowReadingCdkParameters",
-          effect: iam.Effect.ALLOW,
-          actions: ["ssm:GetParameter"],
-          resources: [
-            this.formatArn({
-              service: "ssm",
-              resource: "parameter",
-              resourceName: `cdk-bootstrap/${DEFAULT_BOOSTRAP_QUALIFIER}/*`,
-            }),
-          ],
-        }),
-        // Developers should be able to _read_ from queues within the account (but not modify
-        // the contents). Specific queues and further restrict this access by modifiying the
-        // Queue permissions/policy.
-        new iam.PolicyStatement({
-          sid: "AllowReadingQueues",
-          effect: iam.Effect.ALLOW,
-          actions: ["sqs:Get*", "sqs:List*", "sqs:ReceiveMessage"],
-          resources: ["*"],
-        }),
+        new statement.S3()
+          .allActions()
+          .onBucket(`cdk-*-assets-${cdk.Aws.ACCOUNT_ID}-*`, cdk.Aws.PARTITION)
+          .onBucket("cdktoolkit-stagingbucket-*", cdk.Aws.PARTITION),
+        // The Lookup, Deploy, and other roles will be used by the CDK and use a particular naming
+        // convention
+        new statement.Sts()
+          .toAssumeRole()
+          .onRole(
+            `cdk-${DEFAULT_BOOSTRAP_QUALIFIER}-*-role-${cdk.Aws.ACCOUNT_ID}-${cdk.Aws.REGION}`,
+            cdk.Aws.ACCOUNT_ID,
+            cdk.Aws.PARTITION
+          ),
+        // Additionally, values under the `cdk-bootstrap/` namespace for the qualifier may be necessary
+        // for the CDK to understand the environment and do a deployment
+        new statement.Ssm()
+          .toGetParameter()
+          .onParameter(
+            `cdk-bootstrap/${DEFAULT_BOOSTRAP_QUALIFIER}/*`,
+            cdk.Aws.ACCOUNT_ID,
+            undefined,
```

Comment:
Do you need to pass `undefined` here because you can't set the region?

Reply:
Correct! We could probably also pass `"*"` but `undefined` is the default value. The methods take positional arguments, not a structure/object with named fields; therefore, because we want to later pass a value for `partition`, we must pass something for `region`. So passing `undefined` is basically the same as not passing anything at all.